interface.go

```diff
@@ -0,0 +1,146 @@
+package hcsshim
+
+import (
+	"io"
+	"time"
+)
+
+// ProcessConfig is used as both the input of Container.CreateProcess
+// and to convert the parameters to JSON for passing onto the HCS
+type ProcessConfig struct {
+	ApplicationName  string
+	CommandLine      string
+	WorkingDirectory string
+	Environment      map[string]string
+	EmulateConsole   bool
+	CreateStdInPipe  bool
+	CreateStdOutPipe bool
+	CreateStdErrPipe bool
+	ConsoleSize      [2]int
+}
+
+type Layer struct {
+	ID   string
+	Path string
+}
+
+type MappedDir struct {
+	HostPath      string
+	ContainerPath string
+	ReadOnly      bool
+}
+
+type HvRuntime struct {
+	ImagePath string `json:",omitempty"`
+}
+
+// ContainerConfig is used as both the input of CreateContainer
+// and to convert the parameters to JSON for passing onto the HCS
```

Comment:
Should this all be exposed externally (ProcessConfig, ContainerConfig, etc.)? We could make an external representation and an internal one.

Pros:
- We choose if we want to expose new properties with a version bump to hcsshim. 
- We can conform to Go style with things such as ID instead of Id, etc. (We can also do this with JSON tags and existing structs)
- Breaking changes at HCS (I don't expect any here, but just in case) will not need to be exposed to clients
- Validation can be performed in hcsshim
- More flexible

Cons:
- More work to transition existing consumers to the new API (!)
- Yet another layer to plumb through new properties to HCS
- Must translate the external representation to the format expected at HCS

I'm in favour of adding an external representation, and eventually autogenerating the internal format HCS expects but that will take a bit more time. Thoughts? @jstarks @jhowardmsft